types/credentials.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) OpenFaaS Author(s). All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package types
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path"
+	"strings"
+)
+
+// BasicAuthCredentials are the user and password strings
+type BasicAuthCredentials struct {
+	User     string
+	Password string
+}
+
+// ReadBasicAuth interface for reading the credentials
+type ReadBasicAuth interface {
+	Read() (error, *BasicAuthCredentials)
+}
+
+// ReadBasicAuthFromDisk defines the path for the secret
+type ReadBasicAuthFromDisk struct {
+	SecretMountPath string
+}
+
+func (r *ReadBasicAuthFromDisk) Read() (*BasicAuthCredentials, error) {
+	var credentials *BasicAuthCredentials
+
+	if len(r.SecretMountPath) == 0 {
+		return nil, fmt.Errorf("invalid SecretMountPath specified for reading secrets")
+	}
+
+	userPath := path.Join(r.SecretMountPath, "basic-auth-user")
```

Comment:
Nice to see

Reply:
I believe this is a verbatim lift/shift from the faas project @rgee0 ?